examples/gno.land/r/gnoland/events/events.gno

```diff
@@ -1,240 +1,316 @@
+	} else {
+		panic(ErrInvalidEndTime)
+	}
 
-</div><!-- end column-->
+	std.Emit(EditingEvent,
+		"id",
+		e.id.String(),
+	)
 
-<div class="column">
+	// If sort order was disrupted, sort again
+	if startTime > 0 {
+		sort.Sort(events)
+	}
+}
 
-### EthCC
+// Rendering
 
-- **Come Meet Us at our Booth**
-- Paris, July 17 - 20, 2023
-- Manfred Touron
+// RenderEventWidget shows up to amt of the latest events to a caller
+func RenderEventWidget(amt int) string {
+	eventNum := len(events)
+	if eventNum == 0 {
+		return "No events."
+	}
 
-[Learn more](https://www.ethcc.io/)
+	if amt > 5 {
+		panic("maximum number of events in widget is " + strconv.Itoa(maxWidgetSize))
+	}
 
-</div><!-- end column-->
+	if amt < 1 {
+		panic("amt needs to be more than 0")
+	}
 
-<div class="column">
+	if amt > eventNum {
+		amt = eventNum
+	}
 
-### Eth Seoul
+	output := ""
 
-- **The Evolution of Smart Contracts: A Journey into Gno.land**
-- Seoul, June 3, 2023
-- Manfred Touron
+	for _, event := range events[eventNum-amt:] {
+		if event.startTime.After(time.Now()) {
+			output += ufmt.Sprintf("- [%s](%s)\n", event.name, event.link)
+		}
+	}
 
-[Learn more](https://2023.ethseoul.org/)
+	return output
+}
 
-</div><!-- end column-->
-<div class="column">
+func renderHome(admin bool) string {
+	output := "# gno.land events\n\n"
 
-### BUIDL Asia
+	if len(events) == 0 {
+		output += "No upcoming or past events."
+		return output
+	}
 
-- **Proof of Contribution in Gno.land**
-- Seoul, June 6, 2023
-- Manfred Touron
+	output += "Below is a list of all upcoming and past gno.land events.\n\n"
+	output += ufmt.Sprintf(time.Now().Format(time.RFC850))
 
-[Learn more](https://www.buidl.asia/)
+	output += "---\n\n"
 
-</div><!-- end column-->
-<div class="column">
+	// Find position of la1test past event
+	pos := sort.Search(len(events), func(i int) bool {
+		return events[i].startTime.Before(time.Now())
+	})
 
-### Game Developer Conference
+	// Add upcoming events
+	output += "## Upcoming events\n\n"
+	output += "<div class='columns-3'>"
 
-- **Side Event: Web3 Gaming Apps Powered by Gno**
-- San Francisco, Mach 23, 2023
-- Jae Kwon
+	// If there are any upcoming events
+	if len(events[:pos]) > 0 {
+		for _, e := range events[:pos] {
+			output += e.Render(admin)
+		}
+	} else {
+		output += "No upcoming events.\n\n"
+	}
 
-[Watch the talk](https://www.youtube.com/watch?v=IJ0xel8lr4c)
+	output += "</div>\n\n"
+	output += "---\n\n"
 
-</div><!-- end column-->
-<div class="column">
+	// Add past events
+	output += "## Past events\n\n"
+	output += "<div class='columns-3'>"
 
-### EthDenver
+	if len(events[pos:]) > 0 {
+		for _, e := range events[pos:] {
+			output += e.Render(admin)
+		}
+	} else {
+		output += "No past events.\n\n"
+	}
 
-- **Side Event: Discover Gno.land**
-- Denver, Feb 24 - Mar 5, 2023
-- Jae Kwon
+	output += "</div>\n\n"
 
-[Watch the talk](https://www.youtube.com/watch?v=IJ0xel8lr4c)
+	output += "---\n\n"
+	output += "#### *All times are local."
```

Comment:
Local to what? I think times are local to where the content is rendered by the VM, not by the location of the event.  Perhaps we need timezone as an input parameter when creating events. Or require times as RFC3339 strings.

Reply:
I've added the timezones and the displayed time will include the timezone in UTC, for example UTC+2 or similar.